T4MVCHostMvcApp/T4MVC Files/HomeController.generated.cs

```diff
@@ -521,12 +564,65 @@ public override System.Web.Mvc.ActionResult ActionWithSomeOptionalParams(string
+            if (cancellationToken != default(System.Threading.CancellationToken))
+                ModelUnbinderHelpers.AddRouteValues(callInfo.RouteValueDictionary, "cancellationToken", cancellationToken);
+            ActionWithOptionalParamDefaultingToDefaultValueOverride(callInfo, cancellationToken);
+            return callInfo;
+        }
+
+        [NonAction]
+        partial void ActionWithOptionalParamDefaultingToDefaultValueWithFullyQualifiedTypeNameOverride(T4MVC_System_Web_Mvc_ActionResult callInfo, System.Threading.CancellationToken cancellationToken);
+
+        [NonAction]
+        public override System.Web.Mvc.ActionResult ActionWithOptionalParamDefaultingToDefaultValueWithFullyQualifiedTypeName(System.Threading.CancellationToken cancellationToken)
+        {
+            var callInfo = new T4MVC_System_Web_Mvc_ActionResult(Area, Name, ActionNames.ActionWithOptionalParamDefaultingToDefaultValueWithFullyQualifiedTypeName);
+            if (cancellationToken != default(System.Threading.CancellationToken))
+                ModelUnbinderHelpers.AddRouteValues(callInfo.RouteValueDictionary, "cancellationToken", cancellationToken);
+            ActionWithOptionalParamDefaultingToDefaultValueWithFullyQualifiedTypeNameOverride(callInfo, cancellationToken);
+            return callInfo;
+        }
+
+        [NonAction]
+        partial void ActionWithOptionalParamDefaultingToDefaultValueWithSpacesOverride(T4MVC_System_Web_Mvc_ActionResult callInfo, System.Threading.CancellationToken cancellationToken);
+
+        [NonAction]
+        public override System.Web.Mvc.ActionResult ActionWithOptionalParamDefaultingToDefaultValueWithSpaces(System.Threading.CancellationToken cancellationToken)
+        {
+            var callInfo = new T4MVC_System_Web_Mvc_ActionResult(Area, Name, ActionNames.ActionWithOptionalParamDefaultingToDefaultValueWithSpaces);
+            if (cancellationToken != default(System.Threading.CancellationToken))
+                ModelUnbinderHelpers.AddRouteValues(callInfo.RouteValueDictionary, "cancellationToken", cancellationToken);
+            ActionWithOptionalParamDefaultingToDefaultValueWithSpacesOverride(callInfo, cancellationToken);
+            return callInfo;
+        }
+
+        [NonAction]
+        partial void ActionWithOptionalParamDefaultingToConstructedDefaultValueOverride(T4MVC_System_Web_Mvc_ActionResult callInfo, System.Threading.CancellationToken cancellationToken);
+
+        [NonAction]
+        public override System.Web.Mvc.ActionResult ActionWithOptionalParamDefaultingToConstructedDefaultValue(System.Threading.CancellationToken cancellationToken)
+        {
+            var callInfo = new T4MVC_System_Web_Mvc_ActionResult(Area, Name, ActionNames.ActionWithOptionalParamDefaultingToConstructedDefaultValue);
+            if (cancellationToken != new System.Threading.CancellationToken())
```

Comment:
I don't get the `new` case. How could that ever be equal? `new Foo()` is never equal to `new Foo()`.